packages/datagateway-search/src/card/datasetSearchCardView.component.tsx

```diff
@@ -348,57 +432,69 @@ const DatasetCardView = (props: DatasetCardViewProps): React.ReactElement => {
     () =>
       hierarchy !== 'dls'
         ? [
-            (dataset: Dataset) => (
+            (dataset: SearchResultSource) => (
               <ActionButtonDiv>
                 <AddToCartButton
                   entityType="dataset"
-                  allIds={data?.map((dataset) => dataset.id) ?? []}
+                  allIds={aggregatedIds}
                   entityId={dataset.id}
                 />
                 <DownloadButton
                   entityType="dataset"
                   entityId={dataset.id}
                   entityName={dataset.name}
                   entitySize={
-                    data ? sizeQueries[data.indexOf(dataset)]?.data ?? -1 : -1
+                    sizeQueries[paginatedSource.indexOf(dataset)]?.data ?? -1
                   }
                 />
               </ActionButtonDiv>
             ),
           ]
         : [
-            (dataset: Dataset) => (
+            (dataset: SearchResultSource) => (
               <AddToCartButton
                 entityType="dataset"
-                allIds={data?.map((dataset) => dataset.id) ?? []}
+                allIds={aggregatedIds}
                 entityId={dataset.id}
               />
             ),
           ],
 
-    [data, hierarchy, sizeQueries]
+    [hierarchy, aggregatedIds, sizeQueries, paginatedSource]
   );
 
   return (
-    <CardView
-      data={data ?? []}
-      totalDataCount={totalDataCount ?? 0}
-      onPageChange={pushPage}
-      onFilter={pushFilter}
-      onSort={handleSort}
-      onResultsChange={pushResults}
-      loadedData={!dataLoading}
-      loadedCount={!countLoading}
-      filters={filters}
-      sort={sort}
-      page={page}
-      results={results}
-      title={title}
-      description={description}
-      information={information}
-      moreInformation={moreInformation}
-      buttons={buttons}
-    />
+    <div>
+      {aborted ? (
+        <Paper>
+          <Typography align="center" variant="h6" component="h6">
+            {t('loading.abort_message')}
+          </Typography>
+        </Paper>
+      ) : (
+        <CardView
+          entityName="Dataset"
+          data={paginatedSource ?? []}
+          totalDataCount={aggregatedIds?.length + (hasNextPage ? 1 : 0) ?? 0}
+          onPageChange={pushPage}
+          onFilter={pushFilter}
+          onSort={handleSort}
+          onResultsChange={pushResults}
+          loadedData={!isLoading}
+          loadedCount={!isLoading}
+          filters={parsedFilters}
+          sort={{}}
```

Comment:
I'm just uninformed but why is data not sortable anymore?

Reply:
I'm not sure either, I think Patrick added the code in 860e70a9eff620efecf887309912363dc6ecc9d5 which seems to be for demoing, so I don't know if it's meant to be "un-sortable" or if it's just there so that the demo works.

Reply:
Apologies, Kenneth, I later clarified this with someone. Data is sortable, it just happens in a different place. It happens in that dropdown list next to the search button. So that handles all that now. This can be resolved